### ISSUE TITLE
Abstract base classes high max_length value

### ISSUE THREAD

**Issue by koeaw** [acdh-oeaw/apis-core-rdf#1477]
I noticed the `CharFields` in our abstract base classes provide unusually high `max_length` values of 4096 when I seemed to remember 255 to be the default. Example:
https://github.com/acdh-oeaw/apis-core-rdf/blob/10c524ef36da57924b738cc92cc230cf5a28a3ec/apis_core/apis_entities/abc.py#L13-L16

According to the Django docs on [CharField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.CharField) and [Character fields](https://docs.djangoproject.com/en/5.1/ref/databases/#character-fields), the value is fine for Postgres, which we use, but could become a problem for MySQL users. (The [MySQL docs](https://dev.mysql.com/doc/refman/8.4/en/char.html) suggest `VARCHAR` values exceeding the limit are truncated.)

AFAIK we don't caution against or rule out using MySQL, so if we want our base classes to be more DB agnostic/inclusive of MySQL, we should probably reduce the `max_lengths` to 255. Alternatively, a general recommendation for Postgres over other databases + stating that we specifically develop with Postgres in mind (for which we could cite the base classes as example) would be helpful.

**Comment by b1rger** [comment]
which default was 255?

**Comment by b1rger** [comment]
Could you please specify which part of the documentation you refer to in this sentence:
> According to the Django docs on [CharField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.CharField) and [Character fields](https://docs.djangoproject.com/en/5.1/ref/databases/#character-fields), the value is fine for Postgres, which we use, but could become a problem for MySQL users.

especially in regards to which parts lead you to the conclusion that this "could become a problem for MySQL users"?

**Comment by koeaw** [comment]
> which default was 255?

Sorry, imprecise wording. Substitute "typically used".

255 is the number I had memorised as (max) safe value for `max_length` for `CharField`. The use of 4096 in our classes confused me so I checked if I'd assumed incorrectly for ages or if 255 was outdated info. A [search](https://github.com/search?q=repo%3Aacdh-oeaw%2Fapis-core-rdf%20255&type=code) for the number in our repo just now turns up results which match the image I had in my head.

Re: could become a problem:

Frst sentence from the "Character fields" section I linked earlier:
> Any fields that are stored with VARCHAR column types may have their max_length restricted to 255 characters if you are using unique=True for the field.

We are not setting the fields `unique=True` in Core, but I wouldn't assume that people who inherit from our classes will leave everything as-is. E.g. we want to [modify the fields to add `verbose_names`](https://github.com/acdh-oeaw/apis-instance-tbf/issues/69) for TBF; someone else may need to add a `unique` constraint (assuming this can be done just as easily?) but not realise they might then also have to modify the `max_length` value. Setting this to 255 would be _a precaution_/us acting in a more user-friendly way.

**Comment by b1rger** [comment]
> E.g. we want to https://github.com/acdh-oeaw/apis-instance-tbf/issues/69 for TBF; someone else may need to add a unique constraint

Interesting... how do you change a field you inherited?

**Comment by b1rger** [comment]
> > Interesting... how do you change a field you inherited?
> 
> I have done this by redefining (overriding?) the inherited fields in my models.
> 
> [Fields inherited from abstract base classes can be overridden with another field or value, or be removed with None.](https://docs.djangoproject.com/en/5.1/topics/db/models/#abstract-base-classes)

Yeah, but isn't this then a new field? It doesn't inherit the attributes of the field you redefined/overwrote (i.e. the `max_length` value)?

**Comment by koeaw** [comment]
Hmm, I thought a model's `__init__` method could be overridden to change field attributes. Though I've admittedly only tried this with `verbose_names` so far, for which I don't care about what happens on the DB level but only the UI side.

**Comment by koeaw** [comment]
If this is not a thing that should be done and therefore not something we want to support, then the 2nd part of my initial suggestion is the only revelant thing left to discuss.

**Comment by koeaw** [comment]
> @koeaw I don't get it, in the MySQL documentation you linked (https://dev.mysql.com/doc/refman/8.4/en/char.html) I can't find any information about truncation after 255 characters. I see a confusing limit of "maximum row size" in MySQL but that's not our problem, MySQL users should know that). What am I missing?

My statements were primarily based on the linked sections in the Django docs. It's possible they are wrong/ambiguous/not reflective of the MySQL docs (or that I misinterpreted them).

**Comment by b1rger** [comment]
Maybe you could give us an example code that leads to the problems you see. I tried using the `__init__` method, but that did not have any effect on the datamodel

**Comment by koeaw** [comment]
> Maybe you could give us an example code that leads to the problems you see.

I'm not using MySQL (I don't think any of us are?), would def. have included more info if I was experiencing problems because of this myself!

I opened the issue based on what I read in the docs; this earlier comment has the chain of events:

> 255 is the number I had memorised as (max) safe value for `max_length` for `CharField`. The use of 4096 in our classes confused me so I checked if I'd assumed incorrectly for ages or if 255 was outdated info. A [search](https://github.com/search?q=repo%3Aacdh-oeaw%2Fapis-core-rdf%20255&type=code) for the number in our repo just now turns up results which match the image I had in my head.

> I tried using the `__init__` method, but that did not have any effect on the datamodel

Ah, interesting. So this more of a "hack", a way to manipulate the UI side of things only, hmm. Thank you.

**Comment by b1rger** [comment]
Oke, thanks! Given that we don't even know **if** there is a problem, I'm closing this
